pure.zsh

```diff
@@ -114,13 +132,15 @@ prompt_pure_preprompt_render() {
 		# only redraw if preprompt has changed
 		[[ "${prompt_pure_last_preprompt}" != "${preprompt}" ]] || return
 
-		# calculate length of preprompt for redraw purposes
-		local preprompt_length=$(prompt_pure_string_length $preprompt)
-		local lines=$(( ($preprompt_length - 1) / $COLUMNS + 1 ))
+		# calculate preprompt length and store it in prompt_pure_preprompt_length
+		prompt_pure_string_length_to_var "${preprompt}" "prompt_pure_preprompt_length"
```

Comment:
You don't need to do anything special. `typeset -g` in `prompt_pure_string_length_to_var` is capable of modifying local parameters within the caller. According to [doc](http://zsh.sourceforge.net/Doc/Release/Shell-Builtin-Commands.html):

> The `-g` (global) flag is treated specially: it means that any resulting parameter will not be restricted to local scope. Note that this does not necessarily mean that the parameter will be global, as the flag will apply to any existing parameter (even if unset) from an enclosing function.

Example:

```zsh
> function inner { typeset -g outer_local=modified; }
> function outer { local outer_local=original; inner; print $outer_local; }
> outer
modified
```